### ISSUE TITLE
can't run flyway withouut jarDirs option

### ISSUE THREAD

**Issue by safareli** [NixOS/nixpkgs#59687]
## Issue description

In my `shell.nix` I have `pkgs.postgresql.lib pkgs.postgresql.out postgresql` in my `buildInputs`. I expected flyway to be available but it wasn't so I have added `flyway` to `buildInputs`. then I was able to execute flyway command but it was throwing error like this:
```
ERROR: Directory for Java Migrations not found: /nix/store/h1q4i8grddrccnjxdcl3hj4qd0r335af-flyway-5.2.4/share/jars
```

after looking at it's nix receipt
https://github.com/NixOS/nixpkgs/blob/42a3276/pkgs/development/tools/flyway/default.nix#L16
I saw the jars was copied to unexpected location for the flyway.

the error was coming from here https://github.com/flyway/flyway/blob/bca3e55/flyway-commandline/src/main/java/org/flywaydb/commandline/Main.java#L373
and the JAR_DIRS is computed by default like this https://github.com/flyway/flyway/blob/bca3e55/flyway-commandline/src/main/java/org/flywaydb/commandline/Main.java#L211

So I have added `-jarDirs=/nix/.../share/flyway/jars`  to `flyway` command I was trying to run and then it worked.

The Issue is that you shouldn't need to add the `-jarDirs`. also why is it not enough to install `postgresql` in order to have the flyway available (pg ships with flyway command-line)?

### Steps to reproduce
```
$ nix-env -i flyway
installing 'flyway-5.2.4'
....
$ flyway -user=postgres -password=password -url=jdbc:postgresql://localhost:5432/app_db -locations=filesystem:core/migrations/app -baselineOnMigrate=true info
ERROR: Directory for Java Migrations not found: /nix/store/h1q4i8grddrccnjxdcl3hj4qd0r335af-flyway-5.2.4/share/jars
$ flyway  -jarDirs=/nix/store/h1q4i8grddrccnjxdcl3hj4qd0r335af-flyway-5.2.4/share/flyway/jars -user=postgres -password=password -url=jdbc:postgresql://localhost:5432/app_db -locations=filesystem:core/migrations/app -baselineOnMigrate=true info
Flyway Community Edition 5.2.4 by Boxfuse
Database: jdbc:postgresql://localhost:5432/app_db (PostgreSQL 10.6)
Schema version: 3.22

+-----------+---------+----------------------------------------+------+---------------------+---------+
| Category  | Version | Description                            | Type | Installed On        | State   |
+-----------+---------+----------------------------------------+------+---------------------+---------+
....
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.
```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.5.0, macOS 10.14.4`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2.1`
 - channels(safareli): `"nixpkgs-19.09pre174594.0c0954781e2"`
 - nixpkgs: `/Users/safareli/.nix-defexpr/channels/nixpkgs`
```

**Comment by safareli** [comment]
My suspicion is to add `--add-flags "-jarDirs=$out/share/flyway/jars" \` here https://github.com/NixOS/nixpkgs/blob/42a3276/pkgs/development/tools/flyway/default.nix#L20
I have just started using it and not sure how to test it yet

**Comment by CMCDragonkai** [comment]
I just hit this error as well. I reported it on: https://github.com/flyway/flyway/issues/2390

Basically we need to change the `default.nix` for flyway so that the `jarDirs` can be set by default to the proper location of `share/flyway/jars`. And then allow the user to override if they need to. I found that if you set `-jarDirs` twice the second invocation will be used. But any wrapper override we use must then be lower precedence than the list here: https://stackoverflow.com/questions/51519491/flywaydb-multiple-config-files-for-migration-is-failing So that even environment variables and config files take precedence.
